templates/template.rb

```diff
@@ -374,6 +374,14 @@ def template(name, write_to: nil)
       write_to ||= File.expand_path("../#{name}", __dir__)
       contents = heading + erb.result_with_hash(locals)
 
+      if (extension == ".c" || extension == ".h") && !contents.ascii_only?
+        # Enforce that we only have ASCII characters here. This is necessary
+        # for some locales that only allow ASCII characters in C source files.
```

Comment:
This comment is not exact.
Maybe "other than UTF-8 locales"?

Reply:
That makes sense! I'll update it.